### ISSUE TITLE
Suggestion: add `noop` function

### ISSUE THREAD

**Issue by leaumar** [krzkaczor/ts-essentials#135]
```typescript
function noop(...args: unknown[]): void {
    void args;
}

// Result:
// const x = noop(0, "", false, {}, []); // undefined
```

Unlike the existing `noop-ts`, this function returns `void`. It also avoids the use of `any` altogether, and does not have unused parameters this way (for this project's linting).

When is it handy:
```typescript
// reassign to real callback later - avoids null placeholder
let callback: (event) => void = noop;
// idem
const callback: (event) => void = mapOfCallbacksForEvents.get("myEvent") ?? noop;

function doSomething(): Promise<void> {
    // "delete" T, return Promise<void>
    return getSomething<T>().then(noop);
    // for when the alternative is
    // `await getSomething(); break;`
    // or
    // `await getSomething(); return undefined;`
    // due to code style
}

try {
   dontCareIfThisFails();
} catch(mootError) {
    // linting does not complain about unused variable, and readers get a clear hint of intention
    noop(mootError);
}
```

**Comment by quezak** [comment]
While a `noop` function can be useful, it seems to me like you're using it for wrong reasons -- as a workaround for linter/style problems that shouldn't be there in the first place or have clearer solutions :smiley: 

1. Using `void args` to silence an unused args warning looks like a dirty hack, also it's a _runtime_ workaround for a _non-runtime_ problem. Linter should be disabled for this line instead -- that's what the `eslint-disable-next-line` comments are for. Alternatively, I'm using a [linter config](https://eslint.org/docs/rules/no-unused-vars#argsignorepattern) that allows arg names prefixed with `_` to be unused. And many projects [actually forbid](https://eslint.org/docs/rules/no-void) using the `void` operator anyway.

2. If I saw `return asyncFn().then(noop)` in a code review I would be confused and ask the author what's the intention (being suspicious of some under-the-hood magic required there). This can be done in a more readable way if you need this pattern frequently, but the exact solution will depend on the author's personal preferences. For example, `return ignoreResult(asyncFn())`, `return asyncFn().then(ignoreResult)`, even `return void (await asyncFn())` (though it looks very hacky and I'm not a fan of the `void` operator), or just make the outer function return `Promise<any>` and just return without any "runtime casting".

3. The `catch` example is also a runtime workaround for a linter-only problem. Instead the linter should be configured to [allow certain error names](https://eslint.org/docs/rules/no-unused-vars#caughterrorsignorepattern) to be unused, and/or you can just add a comment to indicate that it's ignored on purpose: `catch (_err) { // ignore }` or `catch (_ignored) {}`

All that said, a `noop` function is still useful if we decide to evolve towards runtime functions too. @krzkaczor thoughts?

**Comment by leaumar** [comment]
1. the `void args` was just intended for the linting of this project itself, if you'd rather omit it and name it `_args` that works too :)
2. I agree a more semantic solution works better there, but to me personally it still makes sense in a way and doesn't confuse me. But yeah, tastes differ, can't argue. I can't agree with returning `Promise<any>` though, the `any` keyword is the biggest evil of all in TS.
3. I guess

I think the net result is we agree on `noop` being useful as a placeholder for optional callbacks and such?

**Comment by macbem** [comment]
I'd replace `void args` with `_args` and I agree with @quezak that those aren't the best examples, because in this case you're just supressing the linter, but I think that having a `noop` in this lib is a good idea.

**Comment by leaumar** [comment]
I just realized btw that `promise.then(erase)` to blank out return types can be achieved with a simple `import noop as erase from "ts-essentials"`. So I think this `noop` is all I'm proposing.

**Comment by quezak** [comment]
> I just realized btw that `promise.then(erase)` to blank out return types can be achieved with a simple `import noop as erase from "ts-essentials"`. So I think this `noop` is all I'm proposing.

yeah, though I'd still call the renamed import `ignoreResult`, to make it clear to the readers that this function doesn't do anything -- but that's up to the enduser to choose the name then ;)

**Comment by krzkaczor** [comment]
Published as `ts-essentials@6.0.3`
